### ISSUE TITLE
strided_slice leads to unknown shape for Input layer

### ISSUE THREAD

**Issue by ipod825** [tensorflow/tensorflow#27649]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
```bash
conda install tensorflow-gpu==2.0-alpha
```
- Python version:
3.7.1
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
not relevent
- GPU model and memory:
not relevent

**Describe the current behavior**
```python
import tensorflow as tf
from tensorflow.keras.layers import Input

with tf.device('/cpu:0'):
    t = Input((2, 3, 4), batch_size=1)
    output = tf.strided_slice(t, [0, 0, 0, 0], [-1, -1, -1, 2])
    print(output.shape)
```
Output is `<unknown>`

**Describe the expected behavior**
strided_slice should calculate the shape of the output tensor correctly.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by sil3nthill** [comment]
hello i would like to work on this and make a PR for it

**Comment by ymodak** [comment]
@ipod825 Thanks for trying TF 2.0 alpha. I was able to reproduce the observed behavior. Executing the same snippet in TF 2.0 cpu version returns correct results.

**Comment by ymodak** [comment]
This is fixed with latest TF 2.0 Nightly. See [gist](https://colab.sandbox.google.com/gist/jvishnuvardhan/c499ffa64c5c541e31a81c876c83fbd9/tf27649.ipynb#scrollTo=wZkzcfDM3TCh)
